### ISSUE TITLE
Identifying TV series does not update to selection and reverts to default provider

### ISSUE THREAD

**Issue by rekyuu** [jellyfin/jellyfin-plugin-anidb#17]
**Describe the bug**

**System (please complete the following information):**
 - OS: Arch Linux
 - Virtualization: None
 - Clients: Browser
 - Browser: Google Chrome 91.0.4472.114
 - Jellyfin Version: 10.7.6-1
 - Playback: N/A
 - Installed Plugins: AniDB 3.0.0.0, AudioDB 10.7.6.0, MusicBrainz 10.7.6.0, OMDb 10.7.6.0
 - Reverse Proxy: None
 - Base URL: None
 - Networking: Host
 - Storage: NFS

**To Reproduce**

1. Click on "Identify" on the series.
2. Search for the show name.
3. Choose the applicable show.

**Expected behavior**

The metadata for the series should update accordingly, but instead selects the topmost selection from the default metadata provider instead. In some cases, the individual seasons update as they should and the main series is incorrectly updated. In other cases, it updates somewhat as you would expect, but episodes don't update and replicate the first episode across all episodes instead.

**Server Logs**

No errors are dumped in the log.

**FFmpeg Logs**

N/A

**Browser Console Logs**
N/A

**Screenshots**

<details>
<summary>Expand</summary>

![2021-07-04T21:15:07 215Z](https://user-images.githubusercontent.com/14059899/124413984-07cc2780-dd0f-11eb-9a9a-171c6a81904e.png)
![2021-07-04T21:15:16 812Z](https://user-images.githubusercontent.com/14059899/124413986-0a2e8180-dd0f-11eb-8380-7eb2b52b2bb2.png)
![2021-07-04T21:15:24 359Z](https://user-images.githubusercontent.com/14059899/124413989-0b5fae80-dd0f-11eb-9bc1-1b902689172f.png)
![2021-07-04T21:15:36 149Z](https://user-images.githubusercontent.com/14059899/124413991-0c90db80-dd0f-11eb-9813-10dcd339b395.png)
![2021-07-04T21:16:30 006Z](https://user-images.githubusercontent.com/14059899/124414005-1286bc80-dd0f-11eb-99b6-354768df8134.png)
![2021-07-04T21:20:15 040Z](https://user-images.githubusercontent.com/14059899/124414008-14508000-dd0f-11eb-8fef-f88ac29ac1b4.png)
</details>

**Additional context**

I've tried multiple plugins using this method, including AniDB, AniList, Kitzu, and TVDB in various orders. In the screenshots above, AniDB is the primary, followed by TheMovieDB.

**Comment by 1337joe** [comment]
Here are my minimal steps to reproduce:

Steps:
1. Install the AniDB plugin.
2. Create a library of type show with at least one show in it.
  -  Ensure the AniDB metadata downloader is enabled as the first metadata downloader in the library (for each category), and also enable TheMovieDb as a secondary downloader.
3. Open the library, select the show, and select "Identify"
4. Enter name "Lost", hit "Search"
5. Select the second entry ("Lost", from TheMovieDb) and hit "Ok"

Result: Metadata (title, images, description, etc) for "Lost Universe" are loaded, and the external links for the show are:
- [AniDB](https://anidb.net/perl-bin/animedb.pl?show=anime&aid=403): Lost Universe
- [IMDb](https://www.imdb.com/title/tt0411008): Lost
- [TheMovieDb](https://www.themoviedb.org/tv/4607): Lost
Expected result: Metadata for the selected show ("Lost") loaded, and AniDB does not have an ID linked because the show isn't listed there.

More details:
The equality check in the series provider passes the same value into both sides, so they can never not be equal. This results in whatever show first matches the search being selected and tagged with an AniDB id.
https://github.com/jellyfin/jellyfin-plugin-anidb/blob/f954e9f71b94fad8aee76266d35b5e2e99256b9a/Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbSeriesProvider.cs#L67

**Comment by rekyuu** [comment]
It's possible AniList uses this same logic, trying to identify A New Hope from TMDb ends up replacing it with the Star Wars: Visions anime. 

The logic I had was that if I put AniList above TMDb it would just tag whatever wasn't anime but it still seems to bug out with it.

**Comment by rekyuu** [comment]
I don't think this is 100% the case with all plugins. I believe it is simply choosing whatever is defined in the default provider for any plguin.

Example with AniList above TMDb, choosing the TMDb metadata:
<details>
<summary>Expand</summary>

![image](https://user-images.githubusercontent.com/14059899/142128840-f69d869f-1b47-45df-abec-c85fd3669cfa.png)

![image](https://user-images.githubusercontent.com/14059899/142128879-35e10017-edab-4f43-bb08-4070d977fde3.png)
</details>

Example with TMDb above AniList, choosing the AniList metadata:

<details>
<summary>Expand</summary>

![image](https://user-images.githubusercontent.com/14059899/142129270-1964694d-7ade-4e42-904a-acf18bb52397.png)

![image](https://user-images.githubusercontent.com/14059899/142129359-dd49f62d-f2ba-45be-ac1c-584814650d4e.png)
</details>

While it's hard to tell between the two providers for Totoro, the description used is from TMDb.

**Comment by 1337joe** [comment]
The steps in your original sequence showed it downloading metadata for a different show than selected by identify because the AniDB provider is overly aggressive on matching. I think that mislead me to what part of the behavior wasn't like you expected.

Identify is (currently) a tool to fix the automatic scanner not picking the right show/movie, not an override for what metadata provider is used for a specific item. Identify simply sets the metadata search fields to match what the selected provider gives then runs that information through the metadata providers as configured for the library.

For instance (using an example from my library that I've been running through this process in the debugger):
1. I use identify to search for "Stargate": I get results for "Stargate SG-1", "Stargate Atlantis", "Stargate Universe", "Stargate Origins", etc.
2. I select the "Stargate SG-1": the TMDb id is immediately saved to the item, then a refresh is triggered using lookup info of the name, production year, and TMDb id (as set in [MetadataService.ApplySearchResult](https://github.com/jellyfin/jellyfin/blob/773a4ae1f34743f1ac80a1661c3061af17d55d8d/MediaBrowser.Providers/Manager/MetadataService.cs#L216)).
3. The refresh is processed by each metadata provider in the order configured in the library settings: if they find a show that matches the lookup info they add their own id and fill in any still-missing metadata (according to their own definition of matching, which is the problem with the AniDB provider).

From your response it sounds more like you expect the flow to be:
1. I use identify to search for "Stargate", get results from various sources.
2. I select "Stargate SG-1" from TheMovieDB: All metadata and images from TheMovieDB are immediately saved to the item and possibly locked so a metadata refresh won't make changes.

I could see a use for that, but it is not currently the way the identify flow was intended or coded. As such, I'd say this is more a feature request than a bug. Should I have this issue moved back to jellyfin/jellyfin and marked as a feature?

**Comment by rekyuu** [comment]
I would agree this should probably be a feature request to make that work. That, or selections from sub-providers should not be available to choose, but I'd prefer the prior.

My current workaround is to change the priority of providers, identify the film or show, then change it back.
